### PR TITLE
Feature/new UI theme - All Subscriptions page, Add New Podcast modal, Podcast Detail page, Episode Info modal

### PR DESCRIPTION
**Pull request by jchue**
### Description

This PR addresses the following UI updates discussed in #158. I have called out specific features I did _not_ complete as part of this, since I first wanted to only do a "reskin" as a first iteration.

- All Subscriptions page (and by extension Favorites page)
  - I have not yet added the new topic filter or the grid/list view toggle.
- Add New Podcast modal
  - I just realized that I have not yet done the PodIndex import. I can do that in the next PR or I can update this one if you prefer.
- Podcast Detail page
  - I have not yet added the episode sort, the duration of each episode, or the checkmark once an episode is complete.
-  Episode Info modal

### Linked Issues

This addresses part but not all of #158.

### Additional context

I'd like to call out that I installed new dependencies for the font and the icons. I've also started to use the icons directly instead of using components. This was because I needed to toggle between the outlined and the filled version (for the "Download to server" indicator), and it would have been messier with a component. Lastly, I created some components for some commonly used elements (headings, buttons, inputs). I hope this is OK.

### CODE REVIEW

**Comment by jchue** [comment]
Sorry, I just noticed that it looks like you use pnpm instead, so we have two different lock files. Should I switch over to pnpm and reinstall the new dependencies?

File: ui/src/pages/PodcastDetailPage.tsx [hunk 1]
```diff
@@ -75,59 +78,63 @@ export const PodcastDetailPage = () => {
...
+                </div>}
+            </div>}
+
             <div>
-                {
-                    selectedEpisodes.map((episode, index) => {
-                        return <PodcastDetailItem episode={episode} key={index} index={index} episodesLength={selectedEpisodes.length}/>
-                    })
-                }
+                <Heading2 className="mb-8">Available Episodes</Heading2>
```
Comment (SamTV12345) [inline ui/src/pages/PodcastDetailPage.tsx:133]: There is a missing translation.

**Comment by SamTV12345** [comment]
Switching to pnpm would be good. Maybe we should switch the default select by a radix drop-down which already contains all the necessary alt attributes + can be styled more easily
(https://www.radix-ui.com/docs/primitives/components/dropdown-menu). I also noticed that the drop-down 1.1.-31.12 isn’t being persisted and defaults to oldest to newest episodes.

**Comment by SamTV12345** [comment]
image
Better use grid than flex box as the box is shorter if someone has a really short description. I think it would be better to have the box take up close to the full width.

**Comment by SamTV12345** [comment]
Other than that I really appreciate your work that you put into the first two pages. If you need any help feel free to comment here.

File: ui/src/pages/Podcasts.tsx [hunk 2]
```diff
@@ -73,52 +73,55 @@ export const Podcasts:FC<PodcastsProps> = ({onlyFavorites})=>{
...
+                    refreshAllPodcasts()
+                }}>refresh</span>
             </div>
-            <button className="bg-amber-300 bg-opacity-70 text-white rounded-xl pl-1 pt-0.5 pr-1" onClick={()=>{
+
+            <ButtonPrimary className="flex items-center" onClick={()=>{
                 dispatch(setModalOpen(true))
-            }}><div className="flex"><PlusIcon/> <span className="p-2 text-amber-700 font-bold">{t('add-new')}</span></div></button>
+            }}>
+                <span className="material-symbols-outlined">add</span> {t('add-new')}
```
[the remark this answers is not in the document]
Comment (jchue) [reply ui/src/pages/Podcasts.tsx:91]: This one isn't displayed text; it's a key for the icon.

File: ui/src/pages/PodcastDetailPage.tsx (hunk 1 above)
Comment (jchue) [reply ui/src/pages/PodcastDetailPage.tsx]: Thanks, fixed. If you can, please double-check the translations, as I just used Google Translate.

**Comment by jchue** [comment]
> Maybe we should switch the default select by a radix drop-down which already contains all the necessary alt attributes + can be styled more easily (https://www.radix-ui.com/docs/primitives/components/select).

Do you want that covered in this iteration? That might also open up other questions about whether we should utilize other components from that library.

> I also noticed that the drop-down 1.1.-31.12 isn’t being persisted and defaults to oldest to newest episodes.

I don't think that's due to the changes I've made. I think it's the same with the latest build.

> The notification is a little bit out of the page
> We should get rid of the burger menu. It should only be displayed on mobile to open the sidebar.

Likewise, I haven't touched the header yet.

**Comment by SamTV12345** [comment]
Ah okay. Forgot that that is also part of the header. Would be good if you could add the dropdown in this iteration so we can say that the pages are done.

**Comment by jchue** [comment]
@SamTV12345 Alright, I've added the Radix select. We might need to work through some merge conflicts, as I noticed there have been changes to the sort/filter in the main branch.

**Comment by SamTV12345** [comment]
Thanks. I fixed the sorting/filter bug in the main branch. Sorry for the misunderstanding I broke that feature 😅.

**Comment by SamTV12345** [comment]
Did the Podindex tab in the add podcast modal look good? Forgot to mention that there is a hidden tab that is only activated once the correct credentials are set.

**Comment by jchue** [comment]
Yep! It looks like it basically shares the same front-end code as the iTunes import, so I don't think I had to do anything additional for it.

image

**Comment by jchue** [comment]
Alright, sure. Do you want me to just keep adding to this PR?

**Comment by SamTV12345** [comment]
Yes. That's alright. I'll merge this one into the new-ui-theme and will try to resolve the merge conflicts.

**Pull request merged**